modules/shared/src/test/scala/org/tessellation/security/signature/SignedValidatorSuite.scala

```diff
@@ -0,0 +1,93 @@
+package org.tessellation.security.signature
+
+import cats.data.{NonEmptySet, Validated}
+import cats.effect.{IO, Resource}
+import cats.syntax.validated._
+
+import org.tessellation.ext.kryo._
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.schema.peer.PeerId
+import org.tessellation.security.signature.Signed.forAsyncKryo
+import org.tessellation.security.{KeyPairGenerator, SecurityProvider}
+import org.tessellation.shared.sharedKryoRegistrar
+
+import eu.timepit.refined.auto._
+import eu.timepit.refined.numeric.Interval
+import weaver.MutableIOSuite
+
+object SignedValidatorSuite extends MutableIOSuite {
+
+  type Res = (KryoSerializer[IO], SecurityProvider[IO])
+
+  override def sharedResource: Resource[IO, SignedValidatorSuite.Res] =
+    KryoSerializer
+      .forAsync[IO](sharedKryoRegistrar.union(Map[Class[_], KryoRegistrationId[Interval.Closed[5000, 5001]]](classOf[TestObject] -> 5000)))
+      .flatMap { ks =>
+        SecurityProvider.forAsync[IO].map((ks, _))
+      }
+
+  test("should succeed when object is signed only by one peer from seedlist") { res =>
+    implicit val (kryo, sp) = res
+
+    for {
+      keyPair1 <- KeyPairGenerator.makeKeyPair[IO]
+      peerId1 = PeerId.fromPublic(keyPair1.getPublic)
+      keyPair2 <- KeyPairGenerator.makeKeyPair[IO]
+      peerId2 = PeerId.fromPublic(keyPair2.getPublic)
+      input = TestObject("Test")
+      signedInput <- forAsyncKryo(input, keyPair1)
+      validator = mkValidator(Some(Set(peerId1, peerId2)))
+      result = validator.validateSignaturesWithSeedlist(signedInput)
+    } yield expect.same(Validated.Valid(signedInput), result)
+  }
+
+  test("should succeed when object is signed only by multiple(2) peers from seedlist") { res =>
```

Comment:
what if there are more than 2 peers?

Reply:
@kriscieslak didn't you mean something like "should only succeed when all signers are on the seedlist"?

Reply:
Changed as suggested.